x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +101,143 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+
+	toDistributeAmount, err := k.ToDistribute.Get(ctx)
+	if err != nil {
+		return sdk.Coin{}, err
+	}
+	if toDistributeAmount.Equal(math.ZeroInt()) {
+		return sdk.Coin{}, nil
+	}
+
+	denom, err := k.iterateAndUpdateFundsDistribution(ctx, toDistributeAmount)
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("error while iterating all the continuous funds: %w", err)
+	}
+
+	// get allocated continuous fund
+	fundsAllocated, err := k.RecipientFundDistribution.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no recipient fund found for recipient: %s", recipient.String())
+		}
+		return sdk.Coin{}, err
+	}
+
+	recipientBal := k.bankKeeper.GetAllBalances(ctx, recipient)
+	recipientAmount := recipientBal.AmountOf(denom)
+	totalRecipientBal := recipientAmount.Add(fundsAllocated)
+	// check if the recipient account balance exceeds maxDistributedCapital after distribution
+	var withdrawnAmount sdk.Coin
+	if totalRecipientBal.LT(cf.MaxDistributedCapital) {
+		withdrawnAmount = sdk.NewCoin(denom, fundsAllocated)
+		// Distribute funds to the recipient from pool module account
+		err := k.DistributeFromCommunityPool(ctx, sdk.NewCoins(withdrawnAmount), recipient)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+
+		// decrement fundsAllocated from ToDistribute
+		toDistribute, err := k.ToDistribute.Get(ctx)
+		if err != nil {
+			return sdk.Coin{}, err
+		}
+		err = k.ToDistribute.Set(ctx, toDistribute.Sub(fundsAllocated))
+		if err != nil {
+			return sdk.Coin{}, err
+		}
```

Comment:
this already happened in iteratefundsdistribute, this should be 0 at this point in the code

Reply:
you're right that this already happened in `iterateAndUpdateFundsDistribution`, but instead this should be removed from here, but not setting to 0. Because `ToDistribute` contains more funds than the distributed amount

Reply:
after distributing to recipent funds it should be 0 as the rest is part of the community pool. Commented above